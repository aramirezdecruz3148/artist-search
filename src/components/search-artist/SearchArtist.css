@import url('https://fonts.googleapis.com/css?family=Monoton|Poiret+One|Viga&display=swap');

.searchArtist {
  display: flex;
  justify-content: center;
  align-items: center;
}

.searchArtist button {
  background-color: rgb(251, 147, 147);
  color: white;
  width: 8vw;
  height: 2vw;
  font-size: 1vw;
  text-align: center;
  padding: 0;
  font-family: 'Viga', sans-serif;
}

.searchArtist button:hover {
  background-color: #AFEEEE;
  transform: scale(1.2);
  color: grey;
}

.searchArtist input {
  width: 14vw;
  height: 1.5vw;
  font-size: 1vw;
  font-family: 'Viga', sans-serif;
  margin-right: .5vw;
}