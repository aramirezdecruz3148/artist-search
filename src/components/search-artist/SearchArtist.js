import React from 'react';
import PropTypes from 'prop-types';

function SearchArtist({ artist, onInputChange }) {
  return (
    <>
      <input type='text' name='artist' value={artist} />
      <button onClick={onInputChange}>Search</button>
    </>
  );
}

SearchArtist.propTypes = {
  artist: PropTypes.string.isRequired,
  onInputChange: PropTypes.func.isRequired
};

export default SearchArtist;
