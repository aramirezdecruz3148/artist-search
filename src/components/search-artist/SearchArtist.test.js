import React from 'react';
import { shallow } from 'enzyme';
import SearchArtist from './SearchArtist';

describe('SearchArtist component', () => {
  it('renders SearchArtist', () => {
    const wrapper = shallow(<SearchArtist artist='me' onButtonClick={() => {}} onInputChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
