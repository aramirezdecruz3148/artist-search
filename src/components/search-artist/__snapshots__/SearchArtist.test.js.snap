// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchArtist component renders SearchArtist 1`] = `
<Fragment>
  <input
    name="artist"
    type="text"
    value="me"
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</Fragment>
`;
